src/Lean/Server/Utils.lean

```diff
@@ -119,7 +126,8 @@ def foldDocumentChanges (changes : @& Array Lsp.TextDocumentContentChangeEvent)
         let minStartOff := minStartOff.min startOff
         ⟨newDocText, minStartOff⟩
       | TextDocumentContentChangeEvent.fullChange (newText : String) =>
-        ⟨newText.toFileMap, 0⟩
+        let newFileMap := newText.toFileMap
+        ⟨newFileMap, firstDiffPos newDocText.source newFileMap.source⟩
```

Comment:
I was briefly confused on reading this code again, maybe `newDocText` wasn't such a good name. Perhaps `accDocText` for "accumulating" is better?

Reply:
Good point.  I've refactored it and extracted the change application logic into a separate function, which avoids the variable naming issue.

There was already an issue due to the confusing variable names: `startOff` was computed by resolving the range against the original `oldText`, and not the current version.  (This wouldn't have affected the end result though.)